rook/XMLDiff.py

```diff
@@ -335,17 +334,51 @@ def fail_message(*args):
         if len(a_element) > 0:
           a_string = ET.tostring(a_element)
           if len(a_string) > 80:
-            message.append('Branches in gold not matching test...\n{}'.format(path))
+            message.append(f'Branches in gold not matching test...\n{path}')
           else:
-            message.append('Branches in gold not matching test...\n{} {}'.format(path, a_string))
+            message.append(f'Branches in gold not matching test...\n{path} {a_string}')
         if len(b_element) > 0:
           b_string = ET.tostring(b_element)
           if len(b_string) > 80:
-            message.append('Branches in test not matching gold...\n{}'.format(path))
+            message.append(f'Branches in test not matching gold...\n{path}')
           else:
-            message.append('Branches in test not matching gold...\n{} {}'.format(path, b_string))
+            message.append(f'Branches in test not matching gold...\n{path} {b_string}')
   return (same, message)
 
+
+def ignore_subnodes_from_tree(a_tree, b_tree, ignored_nodes):
+  """
+    Compares two element trees and returns (same,message) where same is true
+      if they are the same, and message is a list of the differences
+    @ In, a_tree, ET.Element.Tree, the first element tree
+    @ In, b_tree, ET.Element.Tree, the second element tree
+    @ In, ignored_nodes, str, address of ignored subnodes of XPath format, e.g.,
+                              `.//parentNode/childNode[@name:<>]/grandchildNode`
+    @ Out, ignore_subnodes_from_tree, (ET.Element.Tree, ET.Element.Tree, bool), (a_tree,
+          b_tree, success) where a_tree and b_tree are the two input element trees which
+          are modified if successful (otherwise returned as is) and success is a boolean noting if
+          removal of ignored nodes was successful
+  """
+  # internal check of ignored nodes xpath nomenclature already conducted by this point
+  for ignored in ignored_nodes:
+    successes = []
+    for tree in [a_tree, b_tree]:
+      found = False
+      parentpath = '/'.join(ignored.split('/')[:-1]) # `.//parentNode/childNode[@name:<>]`
+      childpath = ignored.split('/')[-1]             # `/grandchildNode`
+      parentnode = tree.find(parentpath)
+      if parentnode is not None:
+        childnode = parentnode.find(childpath)
+        if childnode is not None: #simple boolean check doesn't work?
+          parentnode.remove(childnode)
+          found = True
+      successes.append(found)
+    if sum(successes) == 0:
+      print((f"Given XPath {ignored} not found in either tree."))
+      return a_tree, b_tree, False
```

Comment:
@GabrielSoto-INL I still have a question regarding this process. It seems to me user can provided multiple ignored nodes since there is a "for loop at the beginning". If this is the case, there will be an early return when the first ignored node is not found in both tree, and the rest of ignored nodes will be skipped. I think we should check all the ignored nodes and return after that.

Reply:
Hm, should the last if statement just be taken out of the for loop?

Reply:
I have fixed the issue (and renamed some of the variables to hopefully be clearer)